frontend/src/modules/Shared/Shares/ShareEditForm.js

```diff
@@ -0,0 +1,452 @@
+import {
+  Box,
+  Button,
+  CardContent,
+  CircularProgress,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  TextField,
+  Typography
+} from '@mui/material';
+import { Defaults, Pager, ShareStatus } from '../../../design';
+import SendIcon from '@mui/icons-material/Send';
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  addSharedItem,
+  getShareObject,
+  removeSharedItem,
+  revokeItemsShareObject,
+  submitApproval,
+  updateShareRequestReason
+} from '../../Shares/services';
+import { SET_ERROR } from '../../../globalErrors';
+import { DeleteOutlined } from '@mui/icons-material';
+import PropTypes from 'prop-types';
+import { useNavigate } from 'react-router-dom';
+
+const ItemRow = (props) => {
+  const { share, item, onAction, enqueueSnackbar, dispatch, client } = props;
+
+  const whatToDo = () => {
+    if (!item.status) return 'Request';
+    if (item.status === 'Revoke_Succeeded' || item.status === 'PendingApproval')
+      return 'Delete';
+    if (item.status === 'Share_Succeeded' || item.status === 'Revoke_Failed')
+      return 'Revoke';
+    return 'Nothing';
+  };
+
+  const possibleAction = whatToDo();
+
+  const removeShareItem = async () => {
+    const response = await client.mutate(
+      removeSharedItem({ shareItemUri: item.shareItemUri })
+    );
+    if (!response.errors) {
+      enqueueSnackbar('Item added', {
+        anchorOrigin: {
+          horizontal: 'right',
+          vertical: 'top'
+        },
+        variant: 'success'
+      });
+      await onAction();
+    } else {
+      dispatch({ type: SET_ERROR, error: response.errors[0].message });
+    }
+  };
+
+  const revokeShareItem = async () => {
+    const response = await client.mutate(
+      revokeItemsShareObject({
+        input: {
+          shareUri: share.shareUri,
+          itemUris: [item.itemUri]
```

Comment:
this needs to be `[item.shareItemUri]` rather than `[itemUri]` for this to work

Reply:
true